pkg/operator/csi/csistorageclasscontroller/csi_storageclass_controller.go

```diff
@@ -164,3 +178,54 @@ func removeConditionFn(condType string) v1helpers.UpdateStatusFunc {
 		return nil
 	}
 }
+
+// StorageClassStateEvaluator evaluates the StorageClassState in the corresponding
+// ClusterCSIDriver and reconciles the StorageClass according to that policy.
+// If Managed, apply the SC. If Unmanaged, do nothing. If Removed, delete the SC.
+type StorageClassStateEvaluator struct {
+	kubeClient             kubernetes.Interface
+	clusterCSIDriverLister oplisters.ClusterCSIDriverLister
+	operatorClient         v1helpers.OperatorClient
+	eventRecorder          events.Recorder
+}
+
+func NewStorageClassStateEvaluator(
+	kubeClient kubernetes.Interface,
+	clusterCSIDriverLister oplisters.ClusterCSIDriverLister,
+	eventRecorder events.Recorder) *StorageClassStateEvaluator {
+	return &StorageClassStateEvaluator{
+		kubeClient:             kubeClient,
+		clusterCSIDriverLister: clusterCSIDriverLister,
+		eventRecorder:          eventRecorder,
+	}
+}
+
+func (e *StorageClassStateEvaluator) Eval(ctx context.Context, expectedSC *storagev1.StorageClass) error {
```

Comment:
nit: can we rename the function to - `EvalAndApplyStorageClass` ?

Reply:
Done